### PR TITLE
Adding the ability to provide VertxOptions from the environment

### PR DESCRIPTION
**Pull request by Stwissel**
- amended the VertxParameterProvider
- created TestClass

Motivation:

Reopening of #99 after fixing committer identity.

Conformance:

Your commits should be signed and you should have signed the Eclipse Contributor Agreement as explained in https://github.com/eclipse/vert.x/blob/master/CONTRIBUTING.md
Please also make sure you adhere to the code style guidelines: https://github.com/vert-x3/wiki/wiki/Vert.x-code-style-guidelines

### CODE REVIEW

**Comment by vietj** [comment]
can you add documentation @Stwissel ?

**Comment by jponge** [comment]
+1, LGTM once we have documentation

**Comment by Stwissel** [comment]
added adoc update

File: src/main/asciidoc/index.adoc [hunk 1]
```diff
@@ -149,6 +149,16 @@ Generally-speaking, we respect the JUnit extension scoping rules, but here are t
 3. Injecting in a `@BeforeEach` with no parent context or previous `@BeforeAll` injection creates a new instance shared with the corresponding test and `AfterEach` method(s).
 4. When no instance exists before running a test method, an instance is created for that test (and only for that test).
 
+==== Configuring of `Vertx` objects
```
Comment (jponge) [inline src/main/asciidoc/index.adoc:152]:
```suggestion
==== Configuring `Vertx` instances
```

File: src/main/asciidoc/index.adoc [hunk 2]
```diff
@@ -149,6 +149,16 @@ Generally-speaking, we respect the JUnit extension scoping rules, but here are t
...
+
+By default the `Vertx` objects get created with `Vertx.vertx()`, using the default settings for `Vertx`. However you have the ability to configure `VertxOptions` to suit your needs.
+A typical use case would be "extending blocking timeout warning for debugging". To configure the `Vertx` object you must:
+
+1. create a json file with the `VertxOptions` in https://vertx.io/docs/apidocs/io/vertx/core/VertxOptions.html#VertxOptions-io.vertx.core.json.JsonObject-[json format] 
+2. create an environment variable `vertx.parameter.filename` pointing to that file
+
+With these conditions met, the `Vertx` object will be created with the configured options
+
```
Comment (jponge) [inline src/main/asciidoc/index.adoc:161]: Can you add a short snippet of such a file?

**Review by jponge — CHANGES_REQUESTED** [review]

File: src/main/asciidoc/index.adoc (hunk 1 above)
Comment (Stwissel) [reply src/main/asciidoc/index.adoc:152]: The following paragraph (original documentation) uses "`Vertx` objects", so would that need change too?

File: src/main/asciidoc/index.adoc (hunk 2 above)
Comment (Stwissel) [reply src/main/asciidoc/index.adoc:161]: I was looking for some example in the Vert.x core documentation, couldn't find one

Comment (jponge) [reply src/main/asciidoc/index.adoc:161]: It's a good opportunity to add one here :-)

**Review by jponge — APPROVED** [review]

**Comment by jponge** [comment]
Thanks @Stwissel

**Comment by Stwissel** [comment]
Full example would be:
```
{
  "addressResolverOptions" : {
    "cacheMaxTimeToLive" : 2147483647,
    "cacheMinTimeToLive" : 0,
    "cacheNegativeTimeToLive" : 0,
    "maxQueries" : 4,
    "ndots" : 1,
    "optResourceEnabled" : false,
    "queryTimeout" : 5000,
    "rdFlag" : true,
    "rotateServers" : false,
    "roundRobinInetAddress" : false
  },
  "blockedThreadCheckInterval" : 5,
  "blockedThreadCheckIntervalUnit" : "MINUTES",
  "eventBusOptions" : {
    "acceptBacklog" : -1,
    "clientAuth" : "NONE",
    "clusterPingInterval" : 20000,
    "clusterPingReplyInterval" : 20000,
    "clusterPublicPort" : -1,
    "connectTimeout" : 60000,
    "crlPaths" : [ ],
    "crlValues" : [ ],
    "enabledCipherSuites" : [ ],
    "enabledSecureTransportProtocols" : [ "TLSv1", "TLSv1.1", "TLSv1.2" ],
    "idleTimeout" : 0,
    "idleTimeoutUnit" : "SECONDS",
    "logActivity" : false,
    "port" : 0,
    "receiveBufferSize" : -1,
    "reconnectAttempts" : 0,
    "reconnectInterval" : 1000,
    "reuseAddress" : true,
    "reusePort" : false,
    "sendBufferSize" : -1,
    "soLinger" : -1,
    "ssl" : false,
    "sslHandshakeTimeout" : 10,
    "sslHandshakeTimeoutUnit" : "SECONDS",
    "tcpCork" : false,
    "tcpFastOpen" : false,
    "tcpKeepAlive" : false,
    "tcpNoDelay" : true,
    "tcpQuickAck" : false,
    "trafficClass" : -1,
    "trustAll" : true,
    "useAlpn" : false
  },
  "eventLoopPoolSize" : 24,
  "fileSystemOptions" : {
    "classPathResolvingEnabled" : true,
    "fileCacheDir" : "/var/folders/tb/p0grn6ln00g4nxcs2jnplfkh0000gn/T//vertx-cache",
    "fileCachingEnabled" : true
  },
  "haEnabled" : false,
  "haGroup" : "__DEFAULT__",
  "internalBlockingPoolSize" : 20,
  "maxEventLoopExecuteTime" : 2000000000,
  "maxEventLoopExecuteTimeUnit" : "NANOSECONDS",
  "maxWorkerExecuteTime" : 60000000000,
  "maxWorkerExecuteTimeUnit" : "NANOSECONDS",
  "metricsOptions" : {
    "enabled" : false
  },
  "preferNativeTransport" : false,
  "quorumSize" : 1,
  "warningExceptionTime" : 5000000000,
  "warningExceptionTimeUnit" : "NANOSECONDS",
  "workerPoolSize" : 20
}
```

... added sample in adoc

**Comment by jponge** [comment]
> Full example would be:
> 
> ```
> {
>   "addressResolverOptions" : {
>     "cacheMaxTimeToLive" : 2147483647,
>     "cacheMinTimeToLive" : 0,
>     "cacheNegativeTimeToLive" : 0,
>     "maxQueries" : 4,
>     "ndots" : 1,
>     "optResourceEnabled" : false,
>     "queryTimeout" : 5000,
>     "rdFlag" : true,
>     "rotateServers" : false,
>     "roundRobinInetAddress" : false
>   },
>   "blockedThreadCheckInterval" : 5,
>   "blockedThreadCheckIntervalUnit" : "MINUTES",
>   "eventBusOptions" : {
>     "acceptBacklog" : -1,
>     "clientAuth" : "NONE",
>     "clusterPingInterval" : 20000,
>     "clusterPingReplyInterval" : 20000,
>     "clusterPublicPort" : -1,
>     "connectTimeout" : 60000,
>     "crlPaths" : [ ],
>     "crlValues" : [ ],
>     "enabledCipherSuites" : [ ],
>     "enabledSecureTransportProtocols" : [ "TLSv1", "TLSv1.1", "TLSv1.2" ],
>     "idleTimeout" : 0,
>     "idleTimeoutUnit" : "SECONDS",
>     "logActivity" : false,
>     "port" : 0,
>     "receiveBufferSize" : -1,
>     "reconnectAttempts" : 0,
>     "reconnectInterval" : 1000,
>     "reuseAddress" : true,
>     "reusePort" : false,
>     "sendBufferSize" : -1,
>     "soLinger" : -1,
>     "ssl" : false,
>     "sslHandshakeTimeout" : 10,
>     "sslHandshakeTimeoutUnit" : "SECONDS",
>     "tcpCork" : false,
>     "tcpFastOpen" : false,
>     "tcpKeepAlive" : false,
>     "tcpNoDelay" : true,
>     "tcpQuickAck" : false,
>     "trafficClass" : -1,
>     "trustAll" : true,
>     "useAlpn" : false
>   },
>   "eventLoopPoolSize" : 24,
>   "fileSystemOptions" : {
>     "classPathResolvingEnabled" : true,
>     "fileCacheDir" : "/var/folders/tb/p0grn6ln00g4nxcs2jnplfkh0000gn/T//vertx-cache",
>     "fileCachingEnabled" : true
>   },
>   "haEnabled" : false,
>   "haGroup" : "__DEFAULT__",
>   "internalBlockingPoolSize" : 20,
>   "maxEventLoopExecuteTime" : 2000000000,
>   "maxEventLoopExecuteTimeUnit" : "NANOSECONDS",
>   "maxWorkerExecuteTime" : 60000000000,
>   "maxWorkerExecuteTimeUnit" : "NANOSECONDS",
>   "metricsOptions" : {
>     "enabled" : false
>   },
>   "preferNativeTransport" : false,
>   "quorumSize" : 1,
>   "warningExceptionTime" : 5000000000,
>   "warningExceptionTimeUnit" : "NANOSECONDS",
>   "workerPoolSize" : 20
> }
> ```
> 
> ... added sample in adoc. That one only contains changed values

Perhaps that one shall be in the main doc, wdyt @vietj?

**Pull request merged**